### PR TITLE
copr: add non-null support for normal rpn fn

### PR DESCRIPTION
**Pull request by skyzh**
### What problem does this PR solve?

Problem Summary:

Add `nonnull` support for normal rpn function

### What is changed and how it works?

Proposal: [RFC: Using chunk format in coprocessor framework](https://github.com/tikv/rfcs/pull/43/)

Related Issue: #7724

What's Changed:

* Add `nonnull` attribute for `rpn_fn` macro.
* Two arithmetic functions are refactored to use `nonnull` attribute.

### Check List 

Tests 

- Unit test

### CODE REVIEW

**Comment by skyzh** [comment]
@TennyZhuang @breeswish PTAL~ Thanks

File: components/tidb_query_codegen/src/rpn_function.rs [hunk 1]
```diff
@@ -228,6 +228,9 @@ struct RpnFnAttr {
     /// Whether or not the function is a raw varg function. Raw varg function accepts `&[ScalarValueRef]`.
     is_raw_varg: bool,
 
+    /// Whether or not the function needs extra logic on `None` value.
+    is_nonnull: bool,
```
Comment (breeswish) [inline components/tidb_query_codegen/src/rpn_function.rs:232]: I would prefer `non_null` :)

Comment (skyzh) [reply components/tidb_query_codegen/src/rpn_function.rs:232]: Alright! I'll flip "nullable" for all functions in next PR

**Pull request merged**